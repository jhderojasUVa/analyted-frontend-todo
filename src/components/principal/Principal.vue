<script>
import { BackendAPI } from '../../api'
import ListTodo from '../list-todos/ListTodo.vue';
import AddTodo from '../add-todo/Add-todo.vue';

export default {
    name: 'Application',
    data() {
        return {
            todoItems: [],
            success: false,
            loading: true,
        }
    },
    mounted() {
        // use the method provided!
        this.reloadTodos();
    },
    methods: {
        // Because more than one can be using this
        // and we are not using websockets (TODO for you, add it)
        // we need to reload after every call
        async reloadTodos() {
            try {
                this.todoItems = (await BackendAPI.getAllTodo()).data?.data?.rows;
                this.success = true;
                this.loading = false;
            } catch (err) {
                console.error(err);
                this.success = false;
                this.loading = false;
            }
        },
        // responsible of removing a todo
        async deleteTodo(id) {
            try {
                this.success = (await BackendAPI.removeTodo(id)).sucess;
            } catch (err) {
                console.error(err);
                this.success = false;
            }
            // reload
            this.reloadTodos();
        },
        // responsible of adding a todo
        async addTodo(description) {
            try {
                this.success = (await BackendAPI.addNewTodo(description)).sucess;
            } catch (err) {
                console.error(err.error);
                this.success = false;
            }
            // reload
            this.reloadTodos();
        },
        // change a todo
        async changeTodo(data) {
            try {
                this.success = (await BackendAPI.changeTodo(data)).sucess;
            } catch (err) {
                console.error(err.error);
                this.success = false;
            }
            // reload
            this.reloadTodos();
        }
    },
    components: { ListTodo, AddTodo, AddTodo }
}
</script>

<template>
    <h2>Current ToDos</h2>
    <div v-if="loading">
        <p class="loading">Loading data... please wait...</p>
    </div>
    <div v-else>
        <p v-if="!success" class="error">Sorry. There was an error!</p>
        <p v-else>
            <AddTodo @addTodo="addTodo"/>
            <ListTodo :todoItems="todoItems" @deleteTodo="deleteTodo" @descriptionChanged="changeTodo"/>
        </p>
    </div>
</template>

<style scoped>
    h2 {
        margin-bottom: 2em;
        text-align: center;
    }
    .loading {
        padding: 1em;
        margin: 1em 0em;
        border-radius: 5px; 
        background-color: rgba(10,150,10,0.5);
        color: white;

        text-align: center;
    }
    .error {
        padding: 1em;
        margin: 1em 0em;
        border-radius: 5px; 
        background-color: red;
        color: white;

        text-align: center;
    }
</style>
