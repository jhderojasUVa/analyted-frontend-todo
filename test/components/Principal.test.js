import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import Principal from '../../src/components/principal/Principal.vue';

describe('Principal', () => {
    it('renders', () => {
        const props = {
            todoItems: [],
            sucess: false,
            loading: true,
        };

        const wrapper = mount(Principal, { props });

        expect(wrapper).toMatchSnapshot();
    });
    describe.skip('methods', () => {

    });
})
