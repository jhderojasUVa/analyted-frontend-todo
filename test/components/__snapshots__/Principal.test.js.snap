// Vitest Snapshot v1

exports[`Principal > renders 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <h2
        data-v-1cc90e80=""
      >
        Current ToDos
      </h2>
      <div
        data-v-1cc90e80=""
      >
        <p
          class="loading"
          data-v-1cc90e80=""
        >
          Loading data... please wait...
        </p>
      </div>
      
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "addTodo": [Function],
          "changeTodo": [Function],
          "deleteTodo": [Function],
          "hasOwnProperty": [Function],
          "loading": true,
          "reloadTodos": [Function],
          "success": false,
          "todoItems": [],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "addTodo": 4,
            "changeTodo": 4,
            "constructor": 0,
            "deleteTodo": 4,
            "hasOwnProperty": 4,
            "loading": 2,
            "reloadTodos": 4,
            "success": 2,
            "toJSON": 0,
            "todoItems": 2,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {
            "loading": true,
            "sucess": false,
            "todoItems": [],
          },
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {
            "AddTodo": {
              "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/add-todo/Add-todo.vue",
              "__scopeId": "data-v-eb1153dd",
              "data": [Function],
              "methods": {
                "add": [Function],
              },
              "name": "AddTodo",
              "render": [Function],
            },
            "ListTodo": {
              "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/list-todos/ListTodo.vue",
              "__scopeId": "data-v-cc3ce813",
              "components": {
                "TodoItem": {
                  "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/todo-item/TodoItem.vue",
                  "__scopeId": "data-v-46d2f85e",
                  "components": {
                    "InlineEditing": {
                      "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/inline-editing/InlineEditing.vue",
                      "__scopeId": "data-v-b406172e",
                      "methods": {
                        "updateText": [Function],
                      },
                      "mounted": [Function],
                      "name": "InlineEditing",
                      "props": {
                        "data": [Function],
                      },
                      "render": [Function],
                    },
                  },
                  "data": [Function],
                  "emits": [
                    "deleteTodo",
                    "descriptionChanged",
                    "changeCompleted",
                  ],
                  "methods": {
                    "changeComplete": [Function],
                    "deleteTodo": [Function],
                    "descriptionChanged": [Function],
                    "editItem": [Function],
                  },
                  "name": "TodoItem",
                  "props": {
                    "data": [Function],
                  },
                  "render": [Function],
                },
              },
              "emits": [
                "deleteTodo",
                "descriptionChanged",
                "changeCompleted",
              ],
              "methods": {
                "changeCompleted": [Function],
                "deleteTodo": [Function],
                "descriptionChanged": [Function],
              },
              "name": "ListTodo",
              "props": {
                "todoItems": [Function],
              },
              "render": [Function],
            },
          },
          "ctx": {
            "addTodo": [Function],
            "changeTodo": [Function],
            "deleteTodo": [Function],
            "hasOwnProperty": [Function],
            "loading": true,
            "reloadTodos": [Function],
            "success": false,
            "todoItems": [],
          },
          "da": null,
          "data": {
            "loading": true,
            "success": false,
            "todoItems": [],
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": [
            [Function],
          ],
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "addTodo": [Function],
            "changeTodo": [Function],
            "deleteTodo": [Function],
            "hasOwnProperty": [Function],
            "loading": true,
            "reloadTodos": [Function],
            "success": false,
            "todoItems": [],
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Current ToDos",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h2
                  data-v-1cc90e80=""
                >
                  Current ToDos
                </h2>,
                "key": null,
                "patchFlag": -1,
                "props": null,
                "ref": null,
                "scopeId": "data-v-1cc90e80",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h2",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Loading data... please wait...",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="loading"
                      data-v-1cc90e80=""
                    >
                      Loading data... please wait...
                    </p>,
                    "key": null,
                    "patchFlag": -1,
                    "props": {
                      "class": "loading",
                    },
                    "ref": null,
                    "scopeId": "data-v-1cc90e80",
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <div
                  data-v-1cc90e80=""
                >
                  <p
                    class="loading"
                    data-v-1cc90e80=""
                  >
                    Loading data... please wait...
                  </p>
                </div>,
                "key": 0,
                "patchFlag": 0,
                "props": {
                  "key": 0,
                },
                "ref": null,
                "scopeId": "data-v-1cc90e80",
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Loading data... please wait...",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="loading"
                      data-v-1cc90e80=""
                    >
                      Loading data... please wait...
                    </p>,
                    "key": null,
                    "patchFlag": -1,
                    "props": {
                      "class": "loading",
                    },
                    "ref": null,
                    "scopeId": "data-v-1cc90e80",
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <div
                  data-v-1cc90e80=""
                >
                  <p
                    class="loading"
                    data-v-1cc90e80=""
                  >
                    Loading data... please wait...
                  </p>
                </div>,
                "key": 0,
                "patchFlag": 0,
                "props": {
                  "key": 0,
                },
                "ref": null,
                "scopeId": "data-v-1cc90e80",
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": "data-v-1cc90e80",
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(Fragment),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/principal/Principal.vue",
            "__scopeId": "data-v-1cc90e80",
            "components": {
              "AddTodo": {
                "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/add-todo/Add-todo.vue",
                "__scopeId": "data-v-eb1153dd",
                "data": [Function],
                "methods": {
                  "add": [Function],
                },
                "name": "AddTodo",
                "render": [Function],
              },
              "ListTodo": {
                "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/list-todos/ListTodo.vue",
                "__scopeId": "data-v-cc3ce813",
                "components": {
                  "TodoItem": {
                    "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/todo-item/TodoItem.vue",
                    "__scopeId": "data-v-46d2f85e",
                    "components": {
                      "InlineEditing": {
                        "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/inline-editing/InlineEditing.vue",
                        "__scopeId": "data-v-b406172e",
                        "methods": {
                          "updateText": [Function],
                        },
                        "mounted": [Function],
                        "name": "InlineEditing",
                        "props": {
                          "data": [Function],
                        },
                        "render": [Function],
                      },
                    },
                    "data": [Function],
                    "emits": [
                      "deleteTodo",
                      "descriptionChanged",
                      "changeCompleted",
                    ],
                    "methods": {
                      "changeComplete": [Function],
                      "deleteTodo": [Function],
                      "descriptionChanged": [Function],
                      "editItem": [Function],
                    },
                    "name": "TodoItem",
                    "props": {
                      "data": [Function],
                    },
                    "render": [Function],
                  },
                },
                "emits": [
                  "deleteTodo",
                  "descriptionChanged",
                  "changeCompleted",
                ],
                "methods": {
                  "changeCompleted": [Function],
                  "deleteTodo": [Function],
                  "descriptionChanged": [Function],
                },
                "name": "ListTodo",
                "props": {
                  "todoItems": [Function],
                },
                "render": [Function],
              },
            },
            "data": [Function],
            "methods": {
              "addTodo": [Function],
              "changeTodo": [Function],
              "deleteTodo": [Function],
              "reloadTodos": [Function],
            },
            "mounted": [Function],
            "name": "Application",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": {
          "loading": true,
          "ref": "VTU_COMPONENT",
          "sucess": false,
          "todoItems": [],
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/principal/Principal.vue",
          "__scopeId": "data-v-1cc90e80",
          "components": {
            "AddTodo": {
              "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/add-todo/Add-todo.vue",
              "__scopeId": "data-v-eb1153dd",
              "data": [Function],
              "methods": {
                "add": [Function],
              },
              "name": "AddTodo",
              "render": [Function],
            },
            "ListTodo": {
              "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/list-todos/ListTodo.vue",
              "__scopeId": "data-v-cc3ce813",
              "components": {
                "TodoItem": {
                  "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/todo-item/TodoItem.vue",
                  "__scopeId": "data-v-46d2f85e",
                  "components": {
                    "InlineEditing": {
                      "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/inline-editing/InlineEditing.vue",
                      "__scopeId": "data-v-b406172e",
                      "methods": {
                        "updateText": [Function],
                      },
                      "mounted": [Function],
                      "name": "InlineEditing",
                      "props": {
                        "data": [Function],
                      },
                      "render": [Function],
                    },
                  },
                  "data": [Function],
                  "emits": [
                    "deleteTodo",
                    "descriptionChanged",
                    "changeCompleted",
                  ],
                  "methods": {
                    "changeComplete": [Function],
                    "deleteTodo": [Function],
                    "descriptionChanged": [Function],
                    "editItem": [Function],
                  },
                  "name": "TodoItem",
                  "props": {
                    "data": [Function],
                  },
                  "render": [Function],
                },
              },
              "emits": [
                "deleteTodo",
                "descriptionChanged",
                "changeCompleted",
              ],
              "methods": {
                "changeCompleted": [Function],
                "deleteTodo": [Function],
                "descriptionChanged": [Function],
              },
              "name": "ListTodo",
              "props": {
                "todoItems": [Function],
              },
              "render": [Function],
            },
          },
          "data": [Function],
          "methods": {
            "addTodo": [Function],
            "changeTodo": [Function],
            "deleteTodo": [Function],
            "reloadTodos": [Function],
          },
          "mounted": [Function],
          "name": "Application",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": {
    "addTodo": [Function],
    "changeTodo": [Function],
    "deleteTodo": [Function],
    "hasOwnProperty": [Function],
    "loading": true,
    "reloadTodos": [Function],
    "success": false,
    "todoItems": [],
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": ,
}
`;
